### PR TITLE
Add analysis-RMI to SOSD

### PR DESCRIPTION
**Pull request by alhuan**
Adding analysis version of RMI as described in [Dittrich](https://arxiv.org/pdf/2106.16166.pdf ) to SOSD. analysis-rmi submodule is a fork I made of their repository due to filepath conflicts with existing `rmi` directory.

### CODE REVIEW

File: competitors/rmi_alt.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          auto new_rmi_ptr = std::make_unique<RMIType<KeyType, Layer1, Layer2>>(loading_data, layer2_size);
+          rmi_ = std::move(new_rmi_ptr);
+        });
+  }
+
+  std::string name() const { return "RMIAlternate"; }
+
+  std::size_t size() const {
+    return rmi_->size_in_bytes();
+//    RMIType<KeyType, Layer1, Layer2>& ref = const_cast<RMIType<KeyType, Layer1, Layer2>&>(rmi_);
```
Comment (andreaskipf) [inline competitors/rmi_alt.h]: Let's remove the commented lines.

File: competitors/rmi_alt.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  SearchBound EqualityLookup(const KeyType lookup_key) const {
+    auto it = rmi_->search(lookup_key);
+    return (SearchBound){it.lo, it.hi};
+  }
+
+  int variant() const { return variant_num; }
+
+ private:
+  std::unique_ptr<RMIType<KeyType, Layer1, Layer2>> rmi_;
+};
```
Comment (andreaskipf) [inline competitors/rmi_alt.h]: Add new line.

File: competitors/rmi_alt.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#pragma once
```
Comment (andreaskipf) [inline competitors/rmi_alt.h]: Was this code formatted using clang format?

Comment (alhuan) [reply competitors/rmi_alt.h]: Reformatted now

**Review by andreaskipf — APPROVED** [review]
Great!

**Pull request merged**